atomic_reactor/schemas/config.json

```diff
@@ -39,6 +39,408 @@
         }
       },
       "additionalProperties": false
+    },
+    "koji": {
+        "description": "Koji instance",
+        "type": "object",
+        "properties": {
+            "hub_url": {
+                "description": "Koji hub's xmlrpc url",
+                "type": "string"
+            },
+            "root_url": {
+                "description": "Koji's root storage url",
+                "type": "string"
+            },
+            "auth": {
+                "description": "Authentication information",
+                "type": "object",
+                "properties": {
+                    "proxyuser": {
+                        "description": "Login as different user",
+                        "type": "string"
+                    },
+                    "ssl_certs_dir": {
+                        "description": "Path to directory with cert and ca files",
+                        "type": "string"
+                    },
+                    "krb_cache_path": {
+                        "description": "Path to kerberos credential cache file",
+                        "type": "string"
+                    },
+                    "krb_principal": {
+                        "description": "Kerberos principal",
+                        "type": "string"
+                    },
+                    "krb_keytab_path": {
+                        "description": "Location of Kerberos keytab, e.g. FILE:<absolute_path>",
+                        "type": "string"
+                    }
+                },
+                "additionalProperties": false,
+                "anyOf": [
```

Comment:
I realize now that this schema is not doing all it's supposed to be doing. We want to prevent ssl and kerberos auth from being used together. I think this might to the trick:

```javascript
"auth": {
    "description": "Authentication information",
    "type": "object",
    "properties": {
        "proxyuser": {
            "description": "Login as different user",
            "type": "string"
        }
    },
    "additionalProperties": false,
    "anyOf": [
        {
            "properties": {
                "ssl_certs_dir": {
                  "description": "Path to directory with cert and ca files",
                  "type": "string"      
                }
            },
            "required": ["ssl_certs_dir"]
        },
        {
            "properties": {
                "krb_principal": {
                    "description": "Kerberos principal",
                    "type": "string"
                },
                "krb_keytab_path": {
                    "description": "Location of Kerberos keytab, e.g. FILE:<absolute_path>",
                    "type": "string"
                },
                "krb_keytab_path": {
                    "description": "Location of Kerberos keytab, e.g. FILE:<absolute_path>",
                    "type": "string"
                }
            },
            "required": ["krb_principal", "krb_keytab_path"]
        },
        {
        }
    ]
}
```

Reply:
Good catch!

Reply:
ok